File: 3. Recursion/C++/1. Print Name N Times.cpp
/*
Print name n times using Recursion
*/

#include <bits/stdc++.h>
using namespace std;

void printName(int n){
    if(n==0){
        return;
    }
    cout << "My Name is Sudhir Daga" << "\n";
    printName(n-1);
}

int main() {
	// your code goes here
    int n;
    cin >> n;
    printName(n);
	return 0;
}

File: 3. Recursion/Java/1. Print Name N Times.java
/* package codechef; // don't place package name! */

import java.util.*;
import java.lang.*;
import java.io.*;

/* Name of the class has to be "Main" only if the class is public. */

/*
Print name n times using Recursion
*/

class Codechef
{
    public static void printName(int n){
        if(n==0){
            return;
        }
        System.out.println("My Name is Sudhir Daga");
        printName(n-1);
    }
	public static void main (String[] args) throws java.lang.Exception
	{
		// your code goes here
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		printName(n);
	}
}
